### PR TITLE
neomutt: support list in binds.map

### LINKED ISSUE #1245: neomutt binds.map does not support a list
neomutt will happily accept "index,pager" for the map but the module only allows for a single value.

### PR DESCRIPTION
**Pull request by sumnerevans**
Closes #1245

Adds support for specifying programs.neomutt.binds[].map as a list. If
specified as a list, then the binds will be concatenated with a ",".

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by sumnerevans** [comment]
Conflicts with #1882. Will rebase once that gets merged.

File: modules/programs/neomutt.nix [hunk 1]
```diff
@@ -154,11 +156,15 @@ let
     set sidebar_format = '${cfg.sidebar.format}'
   '';
 
-  bindSection = concatMapStringsSep "\n"
-    (bind: ''bind ${bind.map} ${bind.key} "${bind.action}"'') cfg.binds;
+  genBindMapper =
+    let genBindMap = m: if isList m then concatStringsSep "," m else m;
+    in bindType:
+    concatMapStringsSep "\n"
+    (bind: ''${bindType} ${genBindMap bind.map} ${bind.key} "${bind.action}"'');
```
Comment (berbiche) [inline modules/programs/neomutt.nix:163]:
```suggestion
  genBindMapper =
    concatMapStringsSep "\n"
    (bind: ''${bindType} ${concatStringsSep "," (toList bind.map)} ${bind.key} "${bind.action}"'');
```

Comment (berbiche) [reply modules/programs/neomutt.nix:163]: Because `toList [ "a" ] == [ "a" ]`, you can use this shortcut

**Comment by berbiche** [comment]
Thanks for the contribution. Left a suggestion

**Review by teto — APPROVED** [review]

File: modules/programs/neomutt.nix [hunk 2]
```diff
@@ -54,21 +54,23 @@ let
...
+            "generic"
+            "index"
+            "mix"
+            "pager"
+            "pgp"
+            "postpone"
+            "query"
+            "smime"
+          ];
+        in with types; either (enum menus) (listOf (enum menus));
```
[the remark this answers is not in the document]
Comment (sumnerevans) [reply modules/programs/neomutt.nix:73]: OK, I've added a deprecation warning. Do I also need to add something with the state versions to gate the deprecation? Or do something else to keep track of the fact that this needs to be deprecated?

File: modules/programs/neomutt.nix [hunk 3]
```diff
@@ -154,11 +156,16 @@ let
     set sidebar_format = '${cfg.sidebar.format}'
   '';
 
-  bindSection = concatMapStringsSep "\n"
-    (bind: ''bind ${bind.map} ${bind.key} "${bind.action}"'') cfg.binds;
+  genBindMapper = bindType:
+    concatMapStringsSep "\n" (bind:
+      ''
+        ${bindType} ${
+          concatStringsSep "," (toList bind.map)
```
Comment (teto) [inline modules/programs/neomutt.nix:163]: what if the list is empty ?

File: modules/programs/neomutt.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (teto) [reply modules/programs/neomutt.nix:73]: lets not make it more complex than your actual solution. A mention in the release note should be ok.

File: modules/programs/neomutt.nix (hunk 3 above)
Comment (sumnerevans) [reply modules/programs/neomutt.nix:163]: I added an assertion in d1b06a6 to error if it is empty.

File: modules/programs/neomutt.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (sumnerevans) [reply modules/programs/neomutt.nix:73]: I added a note in the release notes in cc65ca1.

**Comment by sumnerevans** [comment]
I rebased on the latest `master` and the tests look like they finally passed.

**Pull request merged**